diff --git a/events/bedwars/src/lib.rs b/events/bedwars/src/lib.rs
index 8fa0cb9..980e513 100644
--- a/events/bedwars/src/lib.rs
+++ b/events/bedwars/src/lib.rs
@@ -143,8 +143,9 @@ pub fn init_game(address: SocketAddr, crypto: Crypto) -> anyhow::Result<()> {
     app.insert_resource(crypto);
     app.add_plugins((HyperionCore, BedwarsPlugin));
     app.world_mut().trigger(SetProxyAddress {
+        proxy: std::env::var("BEDWARS_PROXY_ADDR")
+            .unwrap_or_else(|_| SetProxyAddress::default().proxy),
         server: address.to_string(),
-        ..SetProxyAddress::default()
     });

     app.run();
